components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/BasicClientAuthenticationHandler.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.auth.service.handler.impl;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpHeaders;
+import org.wso2.carbon.identity.auth.service.AuthenticationContext;
+import org.wso2.carbon.identity.auth.service.AuthenticationResult;
+import org.wso2.carbon.identity.auth.service.AuthenticationStatus;
+import org.wso2.carbon.identity.auth.service.exception.AuthenticationFailException;
+import org.wso2.carbon.identity.auth.service.handler.AuthenticationHandler;
+import org.wso2.carbon.identity.auth.service.util.Constants;
+import org.wso2.carbon.identity.core.bean.context.MessageContext;
+import org.wso2.carbon.identity.core.handler.InitConfig;
+import org.wso2.carbon.identity.oauth.IdentityOAuthAdminException;
+import org.wso2.carbon.identity.oauth.common.exception.InvalidOAuthClientException;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+
+import java.nio.charset.Charset;
+
+import static org.wso2.carbon.identity.auth.service.util.AuthConfigurationUtil.isAuthHeaderMatch;
+
+/**
+ * BasicClientAuthenticationHandler is for authenticate the request based on Basic Authentication
+ * using the client credentials.
+ * canHandle method will confirm whether this request can be handled by this authenticator or not.
+ */
+public class BasicClientAuthenticationHandler extends AuthenticationHandler {
+
+    private static final Log log = LogFactory.getLog(BasicClientAuthenticationHandler.class);
+    private final String BASIC_AUTH_HEADER = "Basic";
+
+    @Override
+    public void init(InitConfig initConfig) {
+
+    }
+
+    @Override
+    public String getName() {
+
+        return Constants.BASIC_CLIENT_AUTH_HANDLER;
+    }
+
+    @Override
+    public int getPriority(MessageContext messageContext) {
+
+        return getPriority(messageContext, 99);
```

Comment:
What will be executed if both BasicClientAuthenticationHandler and BasicAuthenticatior are defined for an endpoint?

Reply:
Only BasicClientAuthenticationHandler will be executed as it has the higher priority.